app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingRunnable.java

```diff
@@ -317,44 +299,22 @@ public void killSyncthing() {
                 break;
             }
 
+            int exitCode;
             for (String syncthingPID : syncthingPIDs) {
-                killProcessId(syncthingPID, i > 0);
-            }
-        }
-    }
-
-    /**
-     * Kill a given process ID
-     *
-     * @param force Whether to use a SIGKILL.
-     */
-    private void killProcessId(String id, boolean force) {
-        Process kill = null;
-        DataOutputStream killOut = null;
-        try {
-            kill = Runtime.getRuntime().exec((mUseRoot) ? "su" : "sh");
-            killOut = new DataOutputStream(kill.getOutputStream());
-            if (!force) {
-                killOut.writeBytes("kill -SIGINT " + id + "\n");
-                killOut.writeBytes("sleep 1\n");
-            } else {
-                killOut.writeBytes("sleep 3\n");
-                killOut.writeBytes("kill -SIGKILL " + id + "\n");
-            }
-            killOut.writeBytes("exit\n");
-            killOut.flush();
-            kill.waitFor();
-            Log.i(TAG_KILL, "Killed Syncthing process "+id);
-        } catch (IOException | InterruptedException e) {
-            Log.w(TAG_KILL, "Failed to kill process id "+id, e);
-        } finally {
-            try {
-                if (killOut != null)
-                    killOut.close();
-            } catch (IOException e) {
-                Log.w(TAG_KILL, "Failed close the killOut stream", e);}
-            if (kill != null) {
-                kill.destroy();
+                if (i > 0) {
+                    // Force termination of the process by sending SIGKILL.
+                    SystemClock.sleep(3000);
```

Comment:
Why do we sleep here again?

Reply:
I just took over the original code, so honestly, I don't know exactly. I suspect this was put in to give the process a plus of 3 secs if previous kill attempts failed to end it before force-killing it.  One can argue if this a higher or lower risk of data corruption if something was stuck during the "normal" kill. Looking at the test logs, I always get syncthing normally killed and not force-killed. My aim by this PR was to remove duplicate code for handling the root shells returning exit codes so I intentionally didn't alter existing code more than necessary.